### PR TITLE
Replace unnecessary DM name strings with classes in MultiDMTestCases

### PR DESCRIPTION
**Pull request by quinton-hoole**
This is part of an attempt to make multiDM integration tests sane, by autogenerating an integration test per combination of DM's.

It is based on #803, so only the third commit 1de958c and onward need to be reviewed.  The first two commits will disappear once #803 is merged and rebased against.

All tests succeed: 

```
./gradlew check
BUILD SUCCESSFUL in 1m 22s
55 actionable tasks: 13 executed, 42 up-to-date
```

### CODE REVIEW

File: core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java [hunk 1]
```diff
@@ -132,26 +142,19 @@ private void runTest(String... dmNames) throws Exception {
...
-    private MicroServiceSpec createMultiDMTestSpec(String... names) throws Exception {
-        String testName = names[0];
-        for (int i = 1; i < names.length; i++) {
-            testName += Character.toUpperCase(names[i].charAt(0)) + names[i].substring(1);
-        }
-
+    private MicroServiceSpec createMultiDMTestSpec(Class... dmClasses) throws Exception {
         MicroServiceSpec spec =
                 MicroServiceSpec.newBuilder()
-                        .setName(testName)
```
Comment (AmitRoushan) [inline core/src/integrationTest/java/amino/run/multidm/MultiDMTestCases.java]: IMHO, It is good to have micro service application deployed with different names. I may help in reusing same testcases for running in parallel.

**Comment by quinton-hoole** [comment]
Yes, but not worth lots of messy code.

**Comment by AmitRoushan** [comment]
@quinton-hoole 
May be we can add one more argument in  runTest (***String testcaseName***, Class... dmClasses) and use it for setting microservice name.

**Comment by quinton-hoole** [comment]
It's not actually a test name.  That variable name is inaccurate.  That name is the name of a microservice.  Which is itself not used anywhere.  This code is all rubbish.  To populate that name it would need to be generated somewhere. I don't think that the name is used anywhere.  Unless there is a concrete reason to add the somewhat complex code to generate a unique name for a field that doesn't make sense and is not used anywhere, I think we just remove all the confusing rubbish, as I have tried to do.

**Comment by quinton-hoole** [comment]
Addressed review feedback. Rebased.  Merging.

**Pull request merged**